yams-py/yams-py/Patcher.py

```diff
@@ -0,0 +1,79 @@
+import json
+import os
+import platform
+import shutil
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+# TODO: revise this.
+yams_path = os.fspath(Path(__file__).with_name(name="yams"))
+sys.path.append(yams_path)
+from pythonnet import load
+load("coreclr")
+import clr
+clr.AddReference("YAMS-LIB")
+from YAMS_LIB import Patcher
+
+
+def patch_game(input_path, output_path, patch_data):
+    # Copy to input dir to temp dir first to do operations there
+    progress_update("Copying to temporary path...", -1)
```

Comment:
I would use some values for these callbacks, even if it's just `0`, `1/4`, `2/4`, `3/4` and `1`.